shop/src/main/java/ro/msg/learning/shop/config/StrategyConfig.java

```diff
@@ -0,0 +1,33 @@
+package ro.msg.learning.shop.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.strategy.LocationStrategy;
+import ro.msg.learning.shop.service.strategy.MultipleLocationsStrategy;
+import ro.msg.learning.shop.service.strategy.SingleLocationStrategy;
+
+@Configuration
+@RequiredArgsConstructor
+public class StrategyConfig {
+
+    private final StockRepository stockRepository;
+    private final LocationRepository locationRepository;
+
+    @Value("${strategy}")
+    private int strategy;
+
+    @Bean
+    public LocationStrategy getStrategyType() {
+        if(strategy == 0) {
```

Comment:
Try using enums for the static values of your strategies. Basically, we would have an enum containing the two values for the strategies (eg. enum LocationStrategy with values SINGLE_LOCATION_STRATEGY and MOST_ABUNDANT_STRATEGY).